repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/datastore/RemoteDatastoreServiceTest.groovy

```diff
@@ -0,0 +1,172 @@
+
+        reposeOneEndpoint = "http://localhost:$repose1Port"
+        reposeTwoEndpoint = "http://localhost:$repose2Port"
+        remoteDatastoreEndpoint = "http://localhost:$datastorePort"
+
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(targetPort)
+
+        def launcherAndLog1 = startRepose('repose1', repose1Port, targetPort, datastorePort, true)
+        repose1 = launcherAndLog1.ReposeValveLauncher
+
+        def launcherAndLog2 = startRepose('repose2', repose2Port, targetPort, datastorePort, true)
+        repose2 = launcherAndLog2.ReposeValveLauncher
+
+        def launcherAndLogR = startRepose('remote', remotePort, targetPort, datastorePort, false)
+        remote = launcherAndLogR.ReposeValveLauncher
+
+        waitUntilReadyToServiceRequests(launcherAndLog1.ReposeLogSearch)
+        waitUntilReadyToServiceRequests(launcherAndLog2.ReposeLogSearch)
+        waitUntilReadyToServiceRequests(launcherAndLogR.ReposeLogSearch)
+    }
+
+    def cleanupSpec() {
+        deproxy?.shutdown()
+        repose1?.stop()
+        repose2?.stop()
+        remote?.stop()
+    }
+
+    static def startRepose(String subName, int reposePort, int targetPort, int datastorePort, boolean client) {
+        def testProperties = new TestProperties(RemoteDatastoreServiceTest.canonicalName.replace('.', '/') + "/$subName")
+        testProperties.setReposePort(reposePort)
+        testProperties.setTargetPort(targetPort)
+        def reposeConfigProvider = new ReposeConfigurationProvider(testProperties)
+        def reposeValveLauncher = new ReposeValveLauncher(reposeConfigProvider, testProperties)
+        reposeValveLauncher.enableDebug()
+        def reposeLogSearch = new ReposeLogSearch(testProperties.getLogFile())
+        reposeLogSearch.cleanLog()
+        def params = testProperties.getDefaultTemplateParams()
+        def runDir = testProperties.runDirectory
+        params.put('repose.artifact.directory', "$runDir/artifacts")
+        params.put('datastorePort', datastorePort)
+        reposeValveLauncher.configurationProvider.applyConfigs("common", params)
+        reposeValveLauncher.configurationProvider.applyConfigs("features/services/datastore/remote", params)
+        def type = client ? "client" : "datastore"
+        reposeValveLauncher.configurationProvider.applyConfigs("features/services/datastore/remote/$type", params)
+        reposeValveLauncher.start(false, false, "repose", type)
+        def artifacts = (new File(runDir, "artifacts")).listFiles(
+                ((FilenameFilter) new WildcardFileFilter("filter-bundle-*.ear")))
+        artifacts.each { file ->
+            def fileOrig = new File(testProperties.reposeHome, "artifacts/${file.name}")
+            def fileNew = new File(runDir, "$subName/artifacts/${file.name}")
+            fileNew.parentFile.mkdirs()
+            def fileDest = new FileOutputStream(fileNew)
+            Files.copy(fileOrig.toPath(), fileDest)
+        }
```

Comment:
The `TestProperties` class was updated to support the parallel testing so that `repose.home` from the `test.properties` is not used if the `testRunDirectory` is supplied.

* https://github.com/rackerlabs/repose/blob/64740bf885967304b4422550e342b95e013c766d/repose-aggregator/tests/functional-tests/src/integrationTest/groovy/framework/TestProperties.groovy#L75

So the short answer is that yes they will look in `${repose.home}/artifacts`, but that isn't where you think it is since we override it for each new Repose instance.

With that said, with the current way we extract all of the artifacts, each Repose instance needs it's own copy of the them. I was talking with @adrianjgeorge a little about this yesterday and he was say the two of you found that we didn't even really need to explode them anymore. If that ends up being the case, then this won't be needed anymore.